### PR TITLE
feat: refactor agent-db into new template

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔/✖                                                                        |
| New feature     | ✔/✖                                                                        |
| Breaking change | ✔/✖                                                                        |
| Deprecations    | ✔/✖                                                                        |
| Documentation   | ✔/✖                                                                        |
| Tests added     | ✔/✖                                                                        |
| Chore            | ✔/✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: config/custom-environment-variables.json [hunk 1]
```diff
@@ -38,5 +38,23 @@
         }
       }
     }
+  },
+  "typeOrm": {
+    "host": "DB_HOST",
+    "port": "DB_PORT",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: "__format": "number"

File: helm/templates/_helpers.tpl [hunk 2]
```diff
@@ -89,21 +82,21 @@ Returns the cloud provider docker registry url from global if exists or from the
...
-{{/*
 Returns the cloud provider image pull secret name from global if exists or from the chart's values
 */}}
-{{- define "ts-server-boilerplate.cloudProviderImagePullSecretName" -}}
+{{- define "discrete-agent-db.cloudProviderImagePullSecretName" -}}
 {{- if .Values.global.cloudProvider.imagePullSecretName }}
     {{- .Values.global.cloudProvider.imagePullSecretName -}}
 {{- else if .Values.cloudProvider.imagePullSecretName -}}
     {{- .Values.cloudProvider.imagePullSecretName -}}
+{{- else -}}
```
Comment (asafMasa) [inline helm/templates/_helpers.tpl:92]: is this a bug in boiler plate? if it is open PR

File: tests/integration/layerHistory/layerHistory.spec.ts [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  describe('Happy Path', function () {
+    it('get should return 200 status code and the history', async function () {
+      findOneMock.mockResolvedValue(inProgressHistoryRecord);
+
+      const response = await requestSender.getHistory(historyIdentifier);
+
+      expect(response.status).toBe(httpStatusCodes.OK);
+      expect(response.body).toEqual(inProgressHistoryResponse);
+      expect(findOneMock).toHaveBeenCalledTimes(1);
+      expect(findOneMock).toHaveBeenCalledWith(findRequest);
```
Comment (asafMasa) [inline tests/integration/layerHistory/layerHistory.spec.ts:75]: add  expect(response).toSatisfyApiSpec(); to each integration test

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: config/custom-environment-variables.json (hunk 1 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: resolve: changed to object with "__format": "number" field

File: helm/templates/_helpers.tpl (hunk 2 above)
Comment (ronenkapelian) [reply helm/templates/_helpers.tpl:92]: not bug,
but another bug was discovered and had been fixed

File: tests/integration/layerHistory/layerHistory.spec.ts (hunk 3 above)
Comment (ronenkapelian) [reply tests/integration/layerHistory/layerHistory.spec.ts:75]: added to all integration tests and was validated - open API was updated too satisfy the tests

**Review by asafMasa — APPROVED** [review]

**Pull request merged**